src/microcalibrate/calibration.py

```diff
@@ -463,3 +487,916 @@ def summary(
         ) / df["Official target"]
         df = df.reset_index(drop=True)
         return df
+
+    def tune_hyperparameters(
```

Comment:
Given how large this function is, and that there are functions within function, I wonder if it would make sense for it to be its own module.